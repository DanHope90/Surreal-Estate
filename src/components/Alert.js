import React from "react";
import PropTypes from "prop-types";

// eslint-disable-next-line react/prop-types
function Alert({ message, success }) {
  if (!message) {
    return null;
  }
  return (
    <div className={`Alert alert-${success ? "success" : "error"}`}>
      {message}
    </div>
  );
  // <div className=`Alert alert-${success ? "success" : "error"}` />
}

export default Alert;

Alert.defaultProps = {
  success: false,
};

Alert.propType = {
  message: PropTypes.string.isRequired,
  success: PropTypes.bool,
};
