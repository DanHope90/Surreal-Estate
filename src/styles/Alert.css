.alert-success, .alert-error{
    width: 144mm;
    height: 25px;
    text-align: center;
    display: flex;
  flex-direction: column;
  align-items: center;



}

.alert-success {
    background-color: lightgreen;
    border-color: black;
    display: flex;
    justify-items: baseline;
}
.alert-error {
    background-color: red;
    border-color: black;

}