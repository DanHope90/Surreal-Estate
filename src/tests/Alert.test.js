import React from "react";
import { render } from "@testing-library/react";
import Alert from "../components/Alert";

describe("Alert", () => {
  it("displays error message", () => {
    const { getByText, asFragment } = render(<Alert message="Error!" />);
    const alert = asFragment();

    expect(getByText(/Error/).textContent).toBe("Error!");
    expect(alert).toMatchSnapshot();
  });

  it("displays success message", () => {
    const { getByText, asFragment } = render(<Alert message="Success!" success />);
    const alert = asFragment();

    expect(getByText(/Success/).textContent).toBe("Success!");
    expect(alert).toMatchSnapshot();
  });

  it("does not render when message is empty", () => {
    const { asFragment } = render(<Alert message="" />);
    const alert = asFragment();

    expect(alert).toMatchSnapshot();
  });
});
