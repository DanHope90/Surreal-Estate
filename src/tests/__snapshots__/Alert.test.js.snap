// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Alert displays error message 1`] = `
<DocumentFragment>
  <div
    class="Alert alert-error"
  >
    Error!
  </div>
</DocumentFragment>
`;

exports[`Alert displays success message 1`] = `
<DocumentFragment>
  <div
    class="Alert alert-success"
  >
    Success!
  </div>
</DocumentFragment>
`;

exports[`Alert does not render when message is empty 1`] = `<DocumentFragment />`;
